docs/source/monitoring.md

```diff
@@ -1,76 +1,180 @@
+200
+```
 
-    ```
-    # Scylla Service Monitor
-    kubectl apply -f examples/common/monitoring/scylla-service-monitor.yaml
+### Connecting through Ingress using an unresolvable domain
 
-    # Scylla Manager Service Monitor
-    kubectl apply -f examples/common/monitoring/scylla-manager-service-monitor.yaml
-    ```
+To connect to an Ingress without a resolvable domain you first need to find out your Ingress controller's IP that can be resolved externally. Again, there are many ways to do so beyond the below examples.
 
-1. Install dashboards
+Unless stated otherwise, we assume your Ingress is running on port 443.
 
-    Scylla Monitoring comes with pre generated dashboards suitable for multiple Scylla versions.
-    In this example we will use dashboards for Scylla 4.6, and Scylla Manager 3.0.
-    Amend directory path to generated dashboards to version suitable for your deployment.
+```console
+$ INGRESS_PORT=443
+```
 
-   Now the dashboards can be created like this:
-   ```
-   # Scylla dashboards
-   for f in scylla-monitoring/grafana/build/ver_4.6/*.json; do
-     kubectl -n scylla-monitoring create configmap scylla-dashboard-"$( basename "${f}" '.json' )" --from-file="${f}" --dry-run=client -o yaml | kubectl label -f- --dry-run=client -o yaml --local grafana_dashboard=1 | kubectl apply --server-side -f-
-   done
+#### Variants
 
-   # Scylla Manager dashboards
-   for f in scylla-monitoring/grafana/build/manager_3.0/*.json; do
-     kubectl -n scylla-monitoring create configmap scylla-manager-dashboard-"$( basename "${f}" '.json' )" --from-file="${f}" --dry-run=client -o yaml | kubectl label -f- --dry-run=client -o yaml --local grafana_dashboard=1 | kubectl apply --server-side -f-
-   done
-    ```
+##### Ingress ExternalIP
 
-    Once Grafana sidecar picks up these dashboards they should be accessible in Grafana.
+When you are running in a real cluster there is usually a cloud LoadBalancer or a bare metal alternative providing you with an externally reachable IP address.
 
-To access Grafana locally, run:
-    ```
-    kubectl -n scylla-monitoring port-forward deployment.apps/monitoring-grafana 3000
-    ```
+```console
+$ INGRESS_IP="$( kubectl -n=haproxy-ingress get service/haproxy-ingress --template='{{ ( index .loadBalancer.ingress 0 ).ip }}' )"
+```
 
-   You can find it on `http://127.0.0.1:3000` and login with the credentials `admin`:`admin`.
\ No newline at end of file
+##### Ingress NodePort
+
+NodePort is slightly less convenient, but it's available in development clusters as well.
+
+```console
+$ INGRESS_IP="$( kubectl get nodes --template='{{ $internal_ip := "" }}{{ $external_ip := "" }}{{ range ( index .items 0 ).status.addresses }}{{ if eq .type "InternalIP" }}{{ $internal_ip = .address }}{{ else if eq .type "ExternalIP" }}{{ $external_ip = .address }}{{ end }}{{ end }}{{ if $external_ip }}{{ $external_ip }}{{ else }}{{ $internal_ip }}{{ end }}' )"
```

Comment:
Shouldn't we prefer the `ExternalIP` first and `InternalIP` as a fallback as it work only if you are in the same network like with your VM but not for say a cloud? The other way around both should work with a fallback.

Reply:
you are right, I've misread while scrolling